pymbolic/typing.py

```diff
@@ -1,33 +1,29 @@
 """
-.. currentmodule:: pymbolic
-
 Typing helpers
 --------------
 
+.. currentmodule:: pymbolic
+
 .. autoclass:: Bool
 .. autoclass:: Number
 .. autoclass:: Scalar
-.. autoclass:: ArithmeticExpression
+.. autodata:: ArithmeticExpression
 
-    A narrower type alias than :class:`Expression` that is returned by
-    arithmetic operators, to allow continue doing arithmetic with the result
-    of arithmetic.
+    A narrower type alias than :class:`~pymbolic.typing.Expression` that is returned
+    by arithmetic operators, to allow continue doing arithmetic with the result.
 
 .. currentmodule:: pymbolic.typing
 
 .. autoclass:: Expression
```

Comment:
Setting this to `autodata` shows the `alias of ...` blurb, but that gives errors in a few places that use `:class:Expression`, so I left it as is.

https://documen.tician.de/pymbolic/primitives.html#pymbolic.typing.Expression

Reply:
The inconsistency really bugged me, so I changed it back and re-added the nitpick ignore.

Reply:
Fair enough, as long as the docs render reasonably well :D 

When looking through this, I found that Sphinx has a nice `missing-reference` event:
https://www.sphinx-doc.org/en/master/extdev/event_callbacks.html#event-missing-reference
that we could hook into and rewire some of these required fake references for maximum hackery :grin:

Reply:
Not ideal:

![grafik](https://github.com/user-attachments/assets/76ceec0b-e520-4efa-ac83-c1efb1d27447)

But could also be worse. I kind of don't want to spend lots of time babysitting sphinx, so this is OK by me. You?

Reply:
Holding up the merge for now to all you to provide feedback.

Reply:
You mean the generated thing for `ArithmeticOrExpressionT`? Yeah, I tried to make that nicer yesterday, but failed miserably :( I'm fine with this as is too.

I was mostly curious about that `<unknown>:1: WARNING` thing and definitely don't want to read through sphinx anymore :(